client/components/image/index.tsx

```diff
@@ -0,0 +1,26 @@
+import classnames from 'classnames';
+import { SyntheticEvent } from 'react';
+
+const hideImageOnError = ( event: SyntheticEvent< HTMLImageElement > ) => {
+	event.currentTarget.style.display = 'none';
+};
+
+export default function Image( { alt, className, src, ...rest }: ImageProps ) {
```

Comment:
You don't need to redefine the `img` tag props, you can use the ones that are exposed by React with `React.ComponentPropsWithoutRef<"img">`

```suggestion
type ImageProps = React.ComponentPropsWithoutRef< 'img' >

export default function Image( { alt, className, src, ...rest }: ImageProps ) {
```